### PR TITLE
Serialize anonymous objects for json body

### PR DESCRIPTION
**Pull request by genlu**
@RikkiGibson @ryzngard

### CODE REVIEW

File: src/GithubMergeTool/GithubMergeTool.cs [hunk 1]
```diff
@@ -187,32 +197,37 @@ git commit
...
-            Console.WriteLine("Creating PR");
-            response = await _client.PostAsyncAsJson($"repos/{repoOwner}/{repoName}/pulls", body);
+            // https://developer.github.com/v3/pulls/#create-a-pull-request
+            response = await _client.PostAsyncAsJson($"repos/{repoOwner}/{repoName}/pulls", JsonConvert.SerializeObject(
+                new
+                {
+                    title = prTitle,
+                    body = prMessage,
+                    head = prBranchName,
+                    @base = destBranch
```
Comment (dibarbet) [inline src/GithubMergeTool/GithubMergeTool.cs]: what does the @ do/mean?

Comment (ryzngard) [reply src/GithubMergeTool/GithubMergeTool.cs]: base is a reserved word, so to name something base you need the `@`

File: src/GithubMergeTool/GithubMergeTool.cs [hunk 2]
```diff
@@ -243,22 +258,17 @@ git commit
...
-                return _client.PostAsyncAsJson(
-                    $"repos/{repoOwner}/{repoName}/git/refs/heads/{branchName}",
-                    $@"
-{{
-    ""sha"": ""{sha}"",
-    ""force"": {(force ? "true" : "false")}
-}}");
+
+                // https://developer.github.com/v3/git/refs/#update-a-reference
+                var body = JsonConvert.SerializeObject( new { sha, force });
```
Comment (RikkiGibson) [inline src/GithubMergeTool/GithubMergeTool.cs]: nit: formatting. should be `SerializeObject(new { ...`

File: src/GithubMergeTool/GithubMergeTool.cs [hunk 3]
```diff
@@ -290,6 +306,7 @@ Task<HttpResponseMessage> PostComment(string prNumber, string comment)
 
             Task<HttpResponseMessage> AddLabels(string prNumber, List<string> labels)
             {
+                // https://developer.github.com/v3/issues/labels/#add-labels-to-an-issue
```
Comment (RikkiGibson) [inline src/GithubMergeTool/GithubMergeTool.cs]: I like these comments, should be useful to maintainers

**Pull request merged**